security/advancedtls/examples/credential_reloading_from_files/README.md

```diff
@@ -7,8 +7,9 @@ This example demonstrates how to set the reloading fields in advancedtls API.
 Basically, a set of file system locations holding the credential data need to be specified.
 Once the credential data needs to be updated, users just change the credential data in the file system, and gRPC will pick up the changes automatically.
 
-This example only shows how to set the API, without demonstrating the way to reload credentials on file system. 
-To learn more about how to do that in Go, please see `advancedtls_integration_test.go`.
+This example demonstrates how to use the `advancedtls` API to reload credentials automatically. 
+This example does not include replacing the credential files with new files. 
+The API consumes those files, but how those files are updated is out of scope.
```

Comment:
Is this actually covered somewhere, then, that we can reference?  This statement seems like it would be very scary to anyone reading it.

Reply:
Yeah, it was covered in `advancedtls_integration_test.go`. I added the previous sentence back. Feel free to let me know how it looks, thanks!

Reply:
Is it fine to do this by just overwriting the files with no ones?  If so, just delete all this stuff and mention "the example doesn't replace the credential files with new files, but if they were overwritten it would read them".

Or...should the example *actually* cover replacing the files, to demonstrate to the user that it works?  It's a "reloading from files" example; if it doesn't actually reload from files, it doesn't seem complete to me.

Reply:
Yes, it's expected to work by overwriting the files with new ones.
The example was original requested by an internal project, and they are going to link the example in their user guide. My original thought for this example was to keep it as simple as enough; @easwars had the same concerns, and my thought was that this example gave instructions on how to use the "reloading" API, instead of demonstrating how "reloading" works. 
I am actually fine with both. We can also overwrite the contents, wait for a while for the client and server to both pick up the new changes, and see if the connection is good, just like what we did in `advancedtls_integration_test.go`. We will need to work on temp files, since the contents are overwritten. But it will make the example a bit more complex, and the same thing seems already covered in `advancedtls_integration_test.go`.
Can I know your opinions on how we should proceed? Thank you so much!

Reply:
I guess, to me, it doesn't seem like an example if it doesn't actually exercise the feature it's claiming to be an example for.  If the example test script can overwrite the files and we can show via logging in the code (e.g.) that the new creds were reloaded properly, that would be ideal.  Even if this is covered by our tests, users don't want to read through hundreds of lines of our test code to learn something simple, so we should not be referring to tests from our documentation.

Please also don't do force pushes unless it's absolutely necessary (e.g. after a rebase).  It makes it impossible to see the history of the PR.  Instead, just make a new commit and push your branch again.  Thanks!

Reply:
OK sounds good! I will modify this PR to add the overwriting behaviors.

Reply:
I've made this PR to show the reloading behaviors on the client side, and log the different common names on server side - I find it very tricky to also reload on the server side, since the timing is hard to control, e.g. It would be pretty hard for the client to know whether the credential has been changed on the server side. 

I previously thought the server could send a message to the client if its credential gets changed, but it didn't work well, because the authentication happened before the rpc call. If the server wants to send some messages to the client, the connection should be established first, but the client has no idea of whether server is using old credentials or new credentials, and the authentication might fail.

But at least we demonstrate the reloading on the client side. Could you please take a look to see how it looks now? Thanks!

(I pushed the commits without `-f` but it still shows `force-update` on the GitHub page. Not sure what happened, but I will keep in mind to force push only when necessary next time)